R/plot.r

```diff
@@ -36,43 +39,40 @@
 #'   evaluation.
 #' @export
 #' @examples
-#' # Generate some sample data, then compute mean and standard deviation
-#' # in each group
+#' # Create a data frame with some sample data, then create a data frame
+#' # containing the mean for each group in the sample data.
 #' set.seed(1)
-#' df <- data.frame(
-#'   gp = factor(rep(letters[1:3], each = 10)),
-#'   y = rnorm(30)
+#' sample_df <- data.frame(
+#'   group = factor(rep(letters[1:3], each = 10)),
+#'   value = rnorm(30)
 #' )
-#' ds <- do.call(rbind, lapply(split(df, df$gp), function(d) {
-#'   data.frame(mean = mean(d$y), sd = sd(d$y), gp = d$gp)
+#' group_means_df <- do.call(rbind, lapply(split(sample_df, sample_df$group), function(d) {
+#'   data.frame(group_mean = mean(d$value), group = d$group)
 #' }))
```

Comment:
Will change this, thanks.

Reply:
Actually, any issues with simplifying the code to create the `group_means_df` to this:

    group_means_df <- sample_df %>%
      group_by(group) %>%
      summarise(group_mean = mean(value))

Reply:
I agree that is more readable and simpler (and exactly why dplyr is convenient), though ggplot2 doesn't import magrittr or dplyr, so we can't use these in examples. The base R pipe is also too new, because it doesn't work on some older R versions which ggplot2 still supports (R 3.5.3). If we don't care about compute an `sd` column, we could also use:

``` r
group_means_df <- setNames(
  aggregate(value ~ group, sample_df, mean), 
  c("group", "group_mean")
)
```

Reply:
Makes sense, and I like your simplified version of the code.
My vote (which I realize carries a fairly low weight) would be to omit the `sd` calculation and the fourth example plot.

Reply:
I'm fine with omitting the 4th example/sd calculation. My comment earlier about the error bars was inquiring, not prescribing.